pkg/cmd/server/origin/run_components.go

```diff
@@ -367,13 +363,118 @@ func (c *MasterConfig) RunDeploymentConfigController() {
+			Reactor:   &triggerdeploymentconfigs.DeploymentConfigReactor{Client: oclient},
+		},
+	}
+	if !c.Options.DisabledFeatures.Has(configapi.FeatureBuilder) {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  unversioned.GroupResource{Group: "build.openshift.io", Resource: "buildconfigs"},
+			Informer:  c.Informers.BuildConfigs().Informer(),
+			Store:     c.Informers.BuildConfigs().Indexer(),
+			TriggerFn: triggerbuildconfigs.NewBuildConfigTriggerIndexer,
+			Reactor:   &triggerbuildconfigs.BuildConfigReactor{Instantiator: bcInstantiator},
+		})
+	}
+	if !c.Options.DisabledFeatures.Has("triggers.image.openshift.io/deployments") {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  unversioned.GroupResource{Group: "extensions", Resource: "deployments"},
+			Informer:  c.Informers.KubernetesInformers().Deployments().Informer(),
+			Store:     c.Informers.KubernetesInformers().Deployments().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+	if !c.Options.DisabledFeatures.Has("triggers.image.openshift.io/daemonsets") {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  unversioned.GroupResource{Group: "extensions", Resource: "daemonsets"},
+			Informer:  c.Informers.KubernetesInformers().DaemonSets().Informer(),
+			Store:     c.Informers.KubernetesInformers().DaemonSets().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+	// TODO: enable post rebase
+	// if !c.Options.DisabledFeatures.Has("triggers.image.openshift.io/statefulsets") {
+	// sources = append(sources, imagetriggercontroller.TriggerSource{
+	// 	Resource:  unversioned.GroupResource{Group: "apps", Resource: "statefulsets"},
+	// 	Informer:  c.Informers.KubernetesInformers().StatefulSets().Informer(),
+	// 	Store:     c.Informers.KubernetesInformers().StatefulSets().Informer().GetIndexer(),
+	// 	TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+	// 	Reactor:  &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+	// })
+	// }
+	// if !c.Options.DisabledFeatures.Has("triggers.image.openshift.io/cronjobs") {
+	// sources = append(sources, imagetriggercontroller.TriggerSource{
+	// 	Resource:  unversioned.GroupResource{Group: "batch", Resource: "cronjobs"},
+	// 	Informer:  c.Informers.KubernetesInformers().CronJobs().Informer(),
+	// 	Store:     c.Informers.KubernetesInformers().CronJobs().Informer().GetIndexer(),
+	// 	TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+	// 	Reactor:  &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+	// })
+	// }
+
+	trigger := imagetriggercontroller.NewTriggerController(
+		broadcaster,
+		streamInformer,
```

Comment:
When we move to shared informers, not exactly.  And passing two interfaces separately is better for testing and composition.